GRIM/EMI: Add Joystick/gamepad support

chkr-private opened the pull request:
This commit adds true joystick/gamepad support to the Grim/Emi engine. I have rebased the commits from #721 and made some minor modifications (comments, support for EMI, ...).

As inherited from ScummVM, the joystick support has to be enabled globally either with the "--joystick" command line option or by adding "joystick_num=0" to the [residualvm] section of ~/.residualvmrc.

Tested with X-Box 360 controller in EMI and GRIM.

klusark commented on common/events.h:
```diff
@@ -87,7 +87,28 @@ enum EventType {
+	EVENT_JOYBUTTON_UP = 25
+};
+
+const int16_t JOYAXIS_MIN = -32768;
```
Should be int16, not int16_t

aquadran commented:
besides above type typo I'm fine this PR

klusark commented:
Maybe add a warning when joystick support is enabled in Grim, but joystick_num is not set?

somaen commented on engines/grim/grim.cpp:
```diff
@@ -119,6 +119,9 @@ GrimEngine::GrimEngine(OSystem *syst, uint32 gameFlags, GrimGameType gameType, C
 		_controlsState[i] = false;
 	}
+	_joyAxisPosition = new float[NUM_JOY_AXES];
+	for (int i = 0; i < NUM_JOY_AXES; i++)
```
Braces on for loop, to avoid mistakes please.

chkr-private replied:
Fixed. Additionally I have also delete(d)[] _joyAxisPositition in ~GrimEngine().

chkr-private commented:
I have addressed all review comments and rebased the branch.

Adding a warning when joystick support is enabled in-game but not in the engine is more complicated: there is no easy way to convey the information from the generic engine/sdl-init into the actual game engine like GrimEngine():
- when residualvm starts, ConfMan contains the "joystick_num" setting either from the command line or from ~/.residualvmrc
- joystick support is enabled in SDL based on this setting
- once a game is selected, all generic config settings in ConfMan are cleared
- once GrimEngine starts up, the status of the common engine's joystick support can't be determined anymore

To avoid any further changes to common code, I'd propose to omit the warning message for now.

somaen commented:
Documentation for end-users would be appreciated.

chkr-private commented:
* rebased again to pick up #1150
* added end user documentation to README.md
